### PR TITLE
remove depracated features

### PR DESCRIPTION
**Pull request by Thomas55555**
Regarding blog:
https://developers.home-assistant.io/blog/2023/07/10/vacuum-updates

### CODE REVIEW

**Comment by prairiesnpr** [comment]
Everything looks good, just not certain I understand the reason for removing the battery and status from the vacuum entity. Is their an upcoming depreciation in HA that I missed?  

Current State:
Screenshot 2023-07-10 at 19 44 12

After Removal:
Screenshot 2023-07-10 at 19 43 49

**Comment by Thomas55555** [comment]
There is a blog post...
I've added it to the description.
And the status attribute is not even documented anymore, although it works.

**Comment by prairiesnpr** [comment]
> There is a blog post... I've added it to the description. And the status attribute is not even documented anymore, although it works.

Got it, not sure I understand the reasoning on HA removing them, but looks like that's where we are.  I'm good, thanks.

**Comment by prairiesnpr** [comment]
We may want to consider retaining the status field as a separate sensor, if we find anyone used it.

**Comment by Thomas55555** [comment]
I think some people use it indeed.
But in the status is only information, which is already somewhere else. I don't think it's necessary to have redundant information.

**Comment by Thomas55555** [comment]
for error, see: https://github.com/home-assistant/developers.home-assistant/pull/1836

**Comment by prairiesnpr** [comment]
Do you mind if we merge #512 then rebase this PR?  Especially since we have now removed the HusqvarnaAutomowerStateMixin.

**Comment by Thomas55555** [comment]
Okay, I'm ready.

File: custom_components/husqvarna_automower/tests/test_vacuum.py [hunk 1]
```diff
@@ -103,104 +102,47 @@ def set_restricted_reason(reason: str):
 
     assert vacuum._attr_unique_id == MWR_ONE_ID
```
Comment (prairiesnpr) [inline custom_components/husqvarna_automower/tests/test_vacuum.py:87]: No need for these tests, should just test the action value

File: custom_components/husqvarna_automower/vacuum.py [hunk 2]
```diff
@@ -27,23 +26,17 @@
 from .const import (
     DOMAIN,
     ERROR_ACTIVITIES,
-    ERROR_STATES,
     ERRORCODES,
```
Comment (prairiesnpr) [inline custom_components/husqvarna_automower/vacuum.py]: Not used

Comment (prairiesnpr) [reply custom_components/husqvarna_automower/vacuum.py]: These should be addressed in the last two commits, just wanted to point them out

File: custom_components/husqvarna_automower/const.py [hunk 3]
```diff
@@ -277,31 +277,5 @@
...
-}
-
-MWR_RES_REASON_TO_STATUS = {
-    "PARK_OVERRIDE": "park_override",
-    "SENSOR": "weather_timer",
-    "DAILY_LIMIT": "daily_limit",
-    "NOT_APPLICABLE": "parked_until_further_notice",
-}
-
+PREV_CONFIG_VER = 2
```
[the remark this answers is not in the document]
Comment (Din-BH) [reply custom_components/husqvarna_automower/const.py]: So.. Is there another way to find this out then? I need to know if the mower had been parked for some time or forever, in order not to override a permanent park with my rain detection autopark automation.

[the remark this answers is not in the document]
Comment (Thomas55555) [reply custom_components/husqvarna_automower/const.py]: It's still in the problem sensor

**Comment by HenrikBachGravesen** [comment]
A comment for the status field.
I've got a switch, which operates 2 guide-wires for my automower. The switch switches between these 2 wires, which opens and closes a pathway from my back lawn to my front lawn.
The path must be closed when the mower is actually mowing, and must be open when the mower is leaving og returning to the dock.
There is an entity "Leaving dock" - if status is no longer available, could I please have an entity for "Returning to dock"?

**Comment by Thomas55555** [comment]
Returning to dock is one of the saturates of the vacuum entity. You can use this. Next time please start a discussion or something like that and don't comment closes PR. It's hard to track stuff here.

**Pull request merged**